api/config/v1alpha1/envoygateway_types.go

```diff
@@ -139,11 +139,23 @@ const (
 	RedisBackendType RateLimitDatabaseBackendType = "Redis"
 )
 
-// RateLimitRedisSettings defines the configuration for connecting to
-// a Redis database.
+// RedisTLSSettings defines the TLS configuration for connecting to redis database.
+type RedisTLSSettings struct {
+	// CertificateRef defines the client certificate reference kubernetes secret for TLS connections.
+	//
+	// +optional
+	CertificateRef *string `json:"certificateRef,omitempty"`
```

Comment:
as shared before, lets keep the `tls` APIs identical
https://github.com/envoyproxy/gateway/blob/main/api/config/v1alpha1/envoygateway_types.go
```
	// CertificateRef contains a references to objects (Kubernetes objects or otherwise) that
	// contains a TLS certificate and private keys. These certificates are used to
	// establish a TLS handshake to the redis server.
	//
	// CertificateRef can only reference a Kubernetes Secret at this time.
	//
	// +kubebuilder:validation:Required
	CertificateRef gwapiv1b1.SecretObjectReference `json:"certificateRef"`
```

Reply:
done